src/enc/impls.rs

```diff
@@ -295,10 +295,17 @@ impl Encode for char {
 
 impl<T> Encode for [T]
 where
-    T: Encode,
+    T: Encode + 'static,
```

Comment:
This change now gives me errors when trying to derive `Encode` for types with lifetimes.

For example:

```rust
struct MyStruct<'a> {
  some_field: &'a String
}
```

`Argument requires that 'a must outlive 'static`

Reply:
Reverted this PR in #663 and tried a better fix in #667